wcs-functions.php

```diff
@@ -149,62 +152,32 @@ function wcs_create_subscription( $args = array() ) {
 		return new WP_Error( 'woocommerce_subscription_invalid_billing_interval', __( 'Invalid subscription billing interval given. Must be an integer greater than 0.', 'woocommerce-subscriptions' ) );
 	}
 
-	$subscription_data['post_type']     = 'shop_subscription';
-	$subscription_data['post_status']   = 'wc-' . apply_filters( 'woocommerce_default_subscription_status', 'pending' );
-	$subscription_data['ping_status']   = 'closed';
-	$subscription_data['post_author']   = 1;
-	$subscription_data['post_password'] = uniqid( 'order_' );
-	// translators: Order date parsed by strftime
-	$post_title_date = strftime( _x( '%b %d, %Y @ %I:%M %p', 'Used in subscription post title. "Subscription renewal order - <this>"', 'woocommerce-subscriptions' ) ); // phpcs:ignore WordPress.WP.I18n.UnorderedPlaceholdersText
-	// translators: placeholder is order date parsed by strftime
-	$subscription_data['post_title']    = sprintf( _x( 'Subscription &ndash; %s', 'The post title for the new subscription', 'woocommerce-subscriptions' ), $post_title_date );
-	$subscription_data['post_date_gmt'] = $args['date_created'];
-	$subscription_data['post_date']     = get_date_from_gmt( $args['date_created'] );
+	$subscription = new \WC_Subscription();
+	$subscription->set_status( $args['status'] );
+	$subscription->set_customer_note( $args['customer_note'] ?? '' );
+	$subscription->set_customer_id( $args['customer_id'] );
+	$subscription->set_date_created( $args['date_created'] );
+	$subscription->set_created_via( $args['created_via'] );
+	$subscription->set_version( $args['order_version'] );
+	$subscription->set_currency( $args['currency'] );
+	$subscription->set_prices_include_tax( $args['prices_include_tax'] );
+	$subscription->set_billing_period( $args['billing_period'] );
+	$subscription->set_billing_interval( absint( $args['billing_interval'] ) );
+	$subscription->set_schedule_start( $args['start_date'] );
```

Comment:
I assume you had to introduce this function because you needed a way to set the start date without calling `$subscription->update_dates()` without saving it. Is that correct?

Reply:
All these setters were already in the function, they have just been reordered. 

https://github.com/Automattic/woocommerce-subscriptions-core/blob/8017438f8e0cfc16494344f783fb134f77284d58/wcs-functions.php#L192-L199

Reply:
Oh yeah I was referring to the `set_schedule_start()` function specifically. That function was introduced in this PR and I was curious as to why it was added instead of using the `update_dates()` function. eg `$subscription->update_dates( [ 'start' => $args['start_date'] ] )`.

I think it was because `update_dates()` actually saves the dates to the database whereas adding this  `set_schedule_start` function just sets the start date meta and it is all saved at once. 

I think this is a good idea, I was just curious if that was the reason behind it.